clients/spark/build.sbt

```diff
@@ -117,7 +120,23 @@ lazy val examples312 = generateExamplesProject(spark312Type).dependsOn(core312)
 lazy val root = (project in file(".")).aggregate(core2, core3, core312, examples2, examples3, examples312)
 
 lazy val assemblySettings = Seq(
-  assembly / assemblyMergeStrategy := (_ => MergeStrategy.first),
+  assembly / assemblyMergeStrategy := {
+    case PathList("META-INF", xs @ _*) =>
+      (xs map {_.toLowerCase}) match {
+        case ("manifest.mf" :: Nil) | ("index.list" :: Nil) | ("dependencies" :: Nil) =>
+          MergeStrategy.discard
+        case ps @ (x :: xs) if ps.last.endsWith(".sf") || ps.last.endsWith(".dsa") =>
+          MergeStrategy.discard
+        case "plexus" :: xs =>
+          MergeStrategy.discard
+        case "services" :: xs =>
+          MergeStrategy.filterDistinctLines
+        case ("spring.schemas" :: Nil) | ("spring.handlers" :: Nil) =>
+          MergeStrategy.filterDistinctLines
+        case _ => MergeStrategy.first
+      }
+    case _ => MergeStrategy.first
+  },
```

Comment:
Can you add documentation for what this does, or where to find its documentation?  I _think_ that it deletes package signing.  If so, are we sure it's OK?

Also, if we say "provided" above do we still need this?

Reply:
This is the default merge strategy of the sbt-assembly plugin with our changes. You can read about it [here](https://github.com/sbt/sbt-assembly#merge-strategy).
Provided doesn't solve the issue...

Reply:
Added the link to docs. 

> I think that it deletes package signing.

My understanding is that it preserves the signing we define in case of a conflict.